File: 1-arrays/easy/1-contains-duplicate-lc-217/swift/contains-duplicate.swift
class ContainsDuplicate {
    func containsDuplicate(_ nums: [Int]) -> Bool {
        var hashSet = Set<Int>()

        for num in nums {
            if hashSet.contains(num) {
                return true
            } else {
                hashSet.insert(num)
            }
        }
        return false
    }
}

File: 1-arrays/easy/1-contains-duplicate-lc-217/rust/contains-duplicate.rs
use std::collections::HashSet;

impl Solution {
    pub fn containsDuplicate(nums: Vec<i32>) -> bool {
        let mut set = HashSet::new();

        for &num in nums.iter() {
            if set.contains(&num) {
                return true;
            } else {
                set.insert(num);
            }
        }
        false
    }
}